miscellaneous_scripts/derive_history.py

```diff
@@ -0,0 +1,69 @@
+import os
+
+archive_folder_path = "/opt/aws/dlc/patch-details-archive"
+latest_patch_details_folder_path = "/opt/aws/dlc/patch-details"
+contents_within_archive_folder = os.listdir(archive_folder_path)
+overall_history_dump_location = os.path.join(
+    os.sep, latest_patch_details_folder_path, "overall_history.txt"
+)
+
+
+def process_patch_folders_and_add_contents(folder_path, overall_history, release_count):
+    """
+    In this method we look into the install scripts within the patch-details folder and extract the contents of the install
+    scripts that are appended to the overall_history.txt
+
+    :param folder_path: str, Path of the patch-details folder
+    :param overall_history: list, contains contents that will finally be dumped into the overall_history.txt file
+    :param release_count: int, stores the release number after the first manual release
+    """
+    if archive_folder_path in folder_path:
+        f = open(os.path.join(os.sep, folder_path, "image_sha.txt"), "r")
+        overall_history.append(f"#### Release {release_count}: {f.read().strip()} ####")
+    elif latest_patch_details_folder_path in folder_path:
+        overall_history.append("#### Latest Release ####")
+    f = open(os.path.join(os.sep, folder_path, "install_script.sh"), "r")
+    overall_history.append(f"{f.read().strip()}")
+    f = open(os.path.join(os.sep, folder_path, "install_script_second.sh"), "r")
+    overall_history.append(f"{f.read().strip()}")
+
+
+def main():
+    """
+    This script is run during the build phase of the AutoPatch DLC and is used by the AutoPatch dockerfile.
+    The script looks into the entire patch history starting from the first release and put the data in the
+    overall_history.txt file.
+    """
+    overall_history = []
+    f = open(os.path.join(os.sep, archive_folder_path, "first_image_sha.txt"), "r")
+    first_image_sha = f.read().strip()
+    overall_history.append(f"#### First image: {first_image_sha} ####")
+    patch_details_folder_names = [
+        artifact_name
+        for artifact_name in contents_within_archive_folder
+        if os.path.isdir(os.path.join(os.sep, archive_folder_path, artifact_name))
+        and artifact_name.startswith("patch-details-")
+    ]
+    patch_details_folder_names = sorted(
+        patch_details_folder_names, key=lambda x: int(x.split("-")[-1])
+    )
+    release_count = 0
+    for patch_details_folder_name in patch_details_folder_names:
```

Comment:
nit: Use `enumerate` to generate an iterator that yields the iteration number and the list-item in the for loop, rather than using a variable that needs to be updated each loop.

https://docs.python.org/3/library/functions.html#enumerate

Reply:
Thanks, used enumerate - done